### PR TITLE
[runtime] Implement wait interrupt in thread info, instead of in the io-layer

### PR DESCRIPTION
**Pull request by ludovic-henry**
This lets us take one more thing out of io-layer, to hopefully reduce it to a bare minimum some day.

### CODE REVIEW

**Comment by ludovic-henry** [comment]
build

**Comment by ludovic-henry** [comment]
build

**Comment by ludovic-henry** [comment]
Comments from @kumpera can be found on https://github.com/ludovic-henry/mono/commit/319f15a9846b698e39cfdd7629fed377e339e4e0

File: mono/io-layer/handles.c [hunk 1]
```diff
@@ -1537,29 +1538,50 @@ static int timedwait_signal_poll_cond (pthread_cond_t *cond, mono_mutex_t *mutex
...
+		gboolean alertable, gboolean poll, gboolean *alerted)
 {
 	DEBUG ("%s: waiting for %p (type %s)", __func__, handle,
 		   _wapi_handle_typename[_wapi_handle_type (handle)]);
-	
+
+	if (alertable)
+		g_assert (alerted);
+
+	if (alerted)
```
Comment (kumpera) [inline mono/io-layer/handles.c]: This conditional is pointless, just init the variable.

**Comment by kumpera** [comment]
Looks good.

**Comment by kumpera** [comment]
Build

**Comment by kumpera** [comment]
build

**Comment by kumpera** [comment]
build

**Pull request merged**